syntax-sloc/raco-sloc.rkt

```diff
@@ -30,20 +31,43 @@
 (define (missing-sloc src)
   (string-append " N/A" "\t" src))
 
+;; lang-line-match : Pregexp -> Path-String -> Boolean
+(define (lang-line-match? px src)
+  (with-input-from-file src
+    (lambda ()
+      (define first-line (for/first ([ln (in-lines)] #:when (< 0 (string-length ln))) ln))
+      (regexp-match? px (string-trim first-line)))))
+
 (module+ main
   (require racket/cmdline)
+  (define *lang-file-pregexp* (make-parameter #f))
   (command-line
    #:program "syntax-sloc"
-   ;; TODO optional arguments
+   #:once-each
+   [("-l" "--lang")
+    lang-pregexp
+    "Only count files with a matching #lang line"
+    (*lang-file-pregexp* (pregexp (string-append "^#lang" "[ \t]*" lang-pregexp "$")))]
```

Comment:
Shouldn't this be "exactly one space" instead of any number of spaces and tabs? That's what actual `#lang` lines provide. And what about `#!racket`? There should be a way to do this properly (and handle comments before it as well) because DrRacket can display the language of a file at the top of the interactions window